### PR TITLE
Loadtestservice - SDK initial review

### PR DESCRIPTION
**Pull request by Yashikaj14**
SDK is being regenerated based on a new swagger spec

### CODE REVIEW

**Comment by jsquire** [comment]
@AlexanderSher:  Would you mind offering thoughts on the generator errors, please?

**Comment by chunyu3** [comment]
> The autorest generation throws following errors on creating the package. error AZC0007: A client type should have a public constructor with equivalent parameters that doesn't take a Azu re.Core.ClientOptions-derived type as the last argument. error AZC0006: A client type should have a public constructor with equivalent parameters that takes a Azure.Core.ClientOp tions-derived type as the last argument How can we resolve this? Any help is appreciated.

Hi @Yashikaj14  @jsquire The root cause may be the invalid definition of Host or Endpoint parameter in swagger json file. I saw in the autorest.md, the json file is an local file so I cannot check the swagger file. Would you please check the swagger file or attach the file in the PR so that we can take a look. Thanks.

**Comment by Yashikaj14** [comment]
> > The autorest generation throws following errors on creating the package. error AZC0007: A client type should have a public constructor with equivalent parameters that doesn't take a Azu re.Core.ClientOptions-derived type as the last argument. error AZC0006: A client type should have a public constructor with equivalent parameters that takes a Azure.Core.ClientOp tions-derived type as the last argument How can we resolve this? Any help is appreciated.
> 
> Hi @Yashikaj14 @jsquire The root cause may be the invalid definition of Host or Endpoint parameter in swagger json file. I saw in the autorest.md, the json file is an local file so I cannot check the swagger file. Would you please check the swagger file or attach the file in the PR so that we can take a look. Thanks.

> Hi @chunyu3, I have updated the swagger file path. Thanks.

**Comment by chunyu3** [comment]
> > > The autorest generation throws following errors on creating the package. error AZC0007: A client type should have a public constructor with equivalent parameters that doesn't take a Azu re.Core.ClientOptions-derived type as the last argument. error AZC0006: A client type should have a public constructor with equivalent parameters that takes a Azure.Core.ClientOp tions-derived type as the last argument How can we resolve this? Any help is appreciated.
> > 
> > 
> > Hi @Yashikaj14 @jsquire The root cause may be the invalid definition of Host or Endpoint parameter in swagger json file. I saw in the autorest.md, the json file is an local file so I cannot check the swagger file. Would you please check the swagger file or attach the file in the PR so that we can take a look. Thanks.
> 
> > Hi @chunyu3, I have updated the swagger file path. Thanks.

Hi @Yashikaj14 
Please remove 'host' definition 
image

and add an parameter 'x-ms-parameterized-host' (see following). You can refer to https://github.com/Azure/azure-rest-api-specs/blob/9d789cce9038ba5b5fc34ffb16c566373b18655c/specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2019-11-01-preview/account.json#L8
image

**Comment by Yashikaj14** [comment]
> > > > The autorest generation throws following errors on creating the package. error AZC0007: A client type should have a public constructor with equivalent parameters that doesn't take a Azu re.Core.ClientOptions-derived type as the last argument. error AZC0006: A client type should have a public constructor with equivalent parameters that takes a Azure.Core.ClientOp tions-derived type as the last argument How can we resolve this? Any help is appreciated.
> > > 
> > > 
> > > Hi @Yashikaj14 @jsquire The root cause may be the invalid definition of Host or Endpoint parameter in swagger json file. I saw in the autorest.md, the json file is an local file so I cannot check the swagger file. Would you please check the swagger file or attach the file in the PR so that we can take a look. Thanks.
> > 
> > 
> > > Hi @chunyu3, I have updated the swagger file path. Thanks.
> 
> Hi @Yashikaj14 Please remove 'host' definition image
> 
> and add an parameter 'x-ms-parameterized-host' (see following). You can refer to https://github.com/Azure/azure-rest-api-specs/blob/9d789cce9038ba5b5fc34ffb16c566373b18655c/specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2019-11-01-preview/account.json#L8 image

Hi @chunyu3, I used that swagger file locally, this change in the swagger has been done and the PR is yet to be merged.

File: sdk/core/Azure.Core.TestFramework/src/Azure.Core.TestFramework.csproj [hunk 1]
```diff
@@ -5,7 +5,6 @@
     <IncludeOperationsSharedSource>true</IncludeOperationsSharedSource>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Azure.Core" />
```
[the remark this answers is not in the document]
Comment (Yashikaj14) [reply sdk/core/Azure.Core.TestFramework/src/Azure.Core.TestFramework.csproj:8]: We reinstalled the package back, so maybe it got removed from there.

**Comment by chunyu3** [comment]
Hi @Yashikaj14 you test recording is not correct. you need to run test live and recording down at the same time, then next time you can run test in recording model.
```
private TestClient CreateClient()
        {
            return InstrumentClient(new TestClient (
                TestEnvironment.Endpoint,
                TestEnvironment.Credential,
                InstrumentClientOptions(new AzureLoadTestingClientOptions())
            ));
        }
```

Have you define the varaible endpoint, credential in environment when you run test in live-model?

**Comment by Yashikaj14** [comment]
> Hi @Yashikaj14 you test recording is not correct. you need to run test live and recording down at the same time, then next time you can run test in recording model.
> 
> ```
> private TestClient CreateClient()
>         {
>             return InstrumentClient(new TestClient (
>                 TestEnvironment.Endpoint,
>                 TestEnvironment.Credential,
>                 InstrumentClientOptions(new AzureLoadTestingClientOptions())
>             ));
>         }
> ```
> 
> Have you define the varaible endpoint, credential in environment when you run test in live-model?

Yes got it, thanks.

File: sdk/loadtestservice/Azure.Analytics.LoadTestService/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Azure LoadTestService client library for .NET
+
+This section should give out brief introduction of the client library.
+
+* First sentence: **Describe the service** briefly. You can usually use the first line of the service's docs landing page for this (Example: [Cosmos DB docs landing page](https://docs.microsoft.com/azure/cosmos-db/)).
+* Next, add a **bulleted list** of the **most common tasks** supported by the package or library, prefaced with "Use the client library for [Product Name] to:". Then, provide code snippets for these tasks in the [Examples](#examples) section later in the document. Keep the task list short but include those tasks most developers need to perform with your package.
+
+  [Source code](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/loadtestservice/Azure.Analytics.LoadTestService/src) | [Package (NuGet)](https://www.nuget.org/packages/Azure.Analytics.LoadTestService) | [API reference documentation](https://azure.github.io/azure-sdk-for-net) | [Product documentation](https://docs.microsoft.com/azure)
```
Comment (pshao25) [inline sdk/loadtestservice/Azure.Analytics.LoadTestService/README.md:8]: @annelo-msft Do we have any release tag? I don't think we should point user to the latest code.

File: sdk/loadtestservice/Azure.Analytics.LoadTestService/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Release History
```
Comment (pshao25) [inline sdk/loadtestservice/Azure.Analytics.LoadTestService/CHANGELOG.md:1]: Add change log

Comment (pshao25) [inline sdk/loadtestservice/Azure.Analytics.LoadTestService/CHANGELOG.md:1]: Add change log.

File: sdk/loadtestservice/Azure.Analytics.LoadTestService/README.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
+# Azure LoadTestService client library for .NET
```
Comment (pshao25) [inline sdk/loadtestservice/Azure.Analytics.LoadTestService/README.md:1]: This file is unfinished. Replace the template content with actual content.

File: sdk/loadtestservice/Azure.Analytics.LoadTestService/src/GlobalSuppressions.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (pshao25) [inline sdk/loadtestservice/Azure.Analytics.LoadTestService/src/GlobalSuppressions.cs:1]: I don't think we should suppress these error. They lack of justification.

File: sdk/loadtestservice/Azure.Analytics.LoadTestService/src/Models/AppComponentModel.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (pshao25) [inline sdk/loadtestservice/Azure.Analytics.LoadTestService/src/Models/AppComponentModel.cs:1]: Why there are models?

File: sdk/loadtestservice/Azure.Analytics.LoadTestService/tests/Azure.Analytics.LoadTestService.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    <PackageReference Include="NUnit" />
+    <PackageReference Include="NUnit3TestAdapter" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" />
+    <PackageReference Include="Moq" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="SessionRecords\" />
+  </ItemGroup>
+</Project>
```
Comment (pshao25) [inline sdk/loadtestservice/Azure.Analytics.LoadTestService/tests/Azure.Analytics.LoadTestService.Tests.csproj:25]: Add a newline. And could you check the all the files have a newline?

File: sdk/loadtestservice/Azure.Analytics.LoadTestService/samples/README.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+---
```
Comment (pshao25) [inline sdk/loadtestservice/Azure.Analytics.LoadTestService/samples/README.md:1]: Could you add your real samples instead of leaving the template?

**Review by pshao25 — CHANGES_REQUESTED** [review]
The template content should be replced by your actual content.

File: sdk/loadtestservice/Azure.Analytics.LoadTestService/src/Models/AppComponentModel.cs (hunk 6 above)
Comment (Yashikaj14) [reply sdk/loadtestservice/Azure.Analytics.LoadTestService/src/Models/AppComponentModel.cs:1]: For easy serialization and deserialization, I created them.

Comment (pshao25) [reply sdk/loadtestservice/Azure.Analytics.LoadTestService/src/Models/AppComponentModel.cs:1]: If it is test purpose, remember to remove it right before the review is ready. If it is customization purpose, please follow https://github.com/Azure/azure-sdk-for-net/blob/e56a0f677e07dd068626af324b8a0f63fa957fcc/doc/DataPlaneCodeGeneration/DeveloperDrivenEvolution.md#L1 to add convenience API.

**Comment by jsquire** [comment]
Closing, per request.

**Pull request closed without merging**